### PR TITLE
feat: support 2fa in the cf tools ui

### PR DESCRIPTION
**Pull request by NetaMolcho**

### CODE REVIEW

**Review by alex-gilin — CHANGES_REQUESTED** [review]
- remove all files that met 'packages/backend/resources/web/*' condition - they are generated resources
- remove packages/frontend/public/generator.png
- remove packages/frontend/src/assets/css/globalStyles.css

File: packages/backend/package.json [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+    "ci": "npm-run-all clean compile copy:test-resources show-out-debug coverage bundle frontend:all package",
+    "ci:artifacts_only": "npm-run-all clean compile webpack-dev frontend:all package",
+    "frontend:all": "cd ../frontend && npm run build && cd ../backend && npm run frontend:copy",
+    "frontend:copy": "npm-run-all clean:frontend && shx cp -r ../frontend/dist/. ./dist/media/",
+    "clean:frontend": "cd dist/ && shx rm -rf ./media",
+    "coverage": "nyc mocha --colors",
+    "bundle": "webpack --mode production",
+    "build-all": "npm run frontend:copy && npm run webpack-dev",
+    "build": "npm-run-all clean compile webpack-dev frontend:all",
+    "copy:test-resources": "cp -r ./tests/resources ./out/tests/resources",
```
Comment (alex-gilin) [inline packages/backend/package.json:31]:
```suggestion
    "copy:test-resources": "shx cp -r ./tests/resources ./out/tests/resources",
```

File: packages/backend/package.json [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+    "clean": "npm-run-all clean:dist clean:out clean:vsix",
+    "clean:dist": "rimraf ./dist",
+    "clean:out": "rimraf ./out",
+    "clean:vsix": "rimraf ./*.vsix",
+    "show-out-debug": "ls -LR -a ./out",
```
Comment (alex-gilin) [inline packages/backend/package.json:36]:
```suggestion
    "show-out-debug": "shx ls -la ./out",
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -13,72 +13,96 @@ jobs: # a collection of steps
...
 workflows:
   version: 2
-  test-deploy:
+  voter:
     jobs:
       - compliance
-      - build-and-test:
+      - build-and-test-node14
+      - build-and-test-node16
+      # - build-and-test-node18
```
[the remark this answers is not in the document]
Comment (veredcon) [reply .circleci/config.yml:92]: removed

File: .eslintrc.js [hunk 4]
```diff
@@ -1,61 +1,69 @@
...
-    "plugin:import/typescript",
-    // prettier must be last because it turns off previous rules
-    "prettier"
-  ],
-  "parser": "@typescript-eslint/parser",
-  "parserOptions": {
-    "project": ["./tsconfig.json"],
-    "tsconfigRootDir": "."
+  rules: {
+    // TODO "eslint-comments/require-description": ["error", { ignore: [] }],
```
[the remark this answers is not in the document]
Comment (veredcon) [reply .eslintrc.js:10]: removed

File: .npmignore [hunk 5]
```diff
@@ -0,0 +1,7 @@
+.circleci
+.vscode
+backend
+frontend
+generator-foodq
```
[the remark this answers is not in the document]
Comment (veredcon) [reply .npmignore:5]: removed .npmignore

File: .vscode/launch.json [hunk 6]
```diff
@@ -1,37 +1,91 @@
...
+      "type": "node",
+      "request": "launch",
+      "name": "Run YoUi websocket server",
+      "program": "${workspaceFolder}/packages/backend/src/webSocketServer/youi.ts",
+      "outFiles": ["${workspaceFolder}/packages/backend/dist/**/*.js"],
+      "env": {
+        "PORT": "8081"
+      },
+      "preLaunchTask": "watch backend (tsc)"
+    },
```
[the remark this answers is not in the document]
Comment (veredcon) [reply .vscode/launch.json:18]: done

File: .vscode/launch.json [hunk 7]
```diff
@@ -1,37 +1,91 @@
...
+      "type": "extensionHost",
+      "request": "launch",
+      "runtimeExecutable": "${execPath}",
+      "args": [
+        "--extensionDevelopmentPath=${workspaceFolder}/packages/backend",
+        "--extensionTestsPath=${workspaceFolder}/packages/backend/dist/test"
+      ],
+      "outFiles": ["${workspaceFolder}/packages/backend/dist/**/*.js"],
+      "preLaunchTask": "watch backend (tsc)"
+    },
```
[the remark this answers is not in the document]
Comment (veredcon) [reply .vscode/launch.json:63]: right - removed

File: .vscode/launch.json [hunk 8]
```diff
@@ -1,37 +1,91 @@
...
+    {
+      "type": "node",
+      "request": "launch",
+      "name": "Run reference generator (foodq)",
+      "program": "${workspaceFolder}/generator-foodq/node_modules/yo/lib/cli.js",
+      "args": ["foodq"],
+      "console": "integratedTerminal",
+      "internalConsoleOptions": "neverOpen",
+      "skipFiles": ["<node_internals>/**"]
+    },
```
[the remark this answers is not in the document]
Comment (veredcon) [reply .vscode/launch.json:73]: done

File: lerna.json [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+  "command": {
+    "publish": {
+      "conventionalCommits": true
+    },
+    "version": {
+      "message": "chore(release): release %s",
+      "allowBranch": "master"
+    }
+  },
+  "version": "0.0.1"
```
[the remark this answers is not in the document]
Comment (veredcon) [reply lerna.json:14]: done

File: packages/backend/package.json [hunk 10]
```diff
@@ -0,0 +1,371 @@
...
+  "main": "./dist/extension.js",
+  "scripts": {
+    "ci": "npm-run-all clean compile copy:test-resources show-out-debug coverage bundle frontend:all package",
+    "ci:artifacts_only": "npm-run-all clean compile webpack-dev frontend:all package",
+    "frontend:all": "cd ../frontend && npm run build && cd ../backend && npm run frontend:copy",
+    "frontend:copy": "npm-run-all clean:frontend && shx cp -r ../frontend/dist/. ./dist/media/",
+    "clean:frontend": "cd dist/ && shx rm -rf ./media",
+    "coverage": "nyc mocha --colors",
+    "bundle": "webpack --mode production",
+    "build-all": "npm run frontend:copy && npm run webpack-dev",
```
[the remark this answers is not in the document]
Comment (veredcon) [reply packages/backend/package.json:29]: removed "build-all"

File: packages/backend/package.json (hunk 1 above)
Comment (veredcon) [reply packages/backend/package.json:31]: done

**Review by alex-gilin — APPROVED** [review]

**Pull request merged**